src/main/java/org/reaktivity/nukleus/maven/plugin/internal/ast/parse/AstParser.java

```diff
@@ -504,9 +525,17 @@ public AstNode visitUnbounded_octets_type(Unbounded_octets_typeContext ctx)
     public AstNode visitScoped_name(
         Scoped_nameContext ctx)
     {
+        String typeName = ctx.getText();
         if (memberBuilder != null)
         {
-            memberBuilder.type(AstType.dynamicType(ctx.getText()));
+            if (typeName.contains("::"))
+            {
+                memberBuilder.type(AstType.dynamicType(typeName));
+            }
+            else if (scopeNameMap.containsKey(typeName))
+            {
+                memberBuilder.type(AstType.dynamicType(scopeNameMap.get(typeName)));
+            }
```

Comment:
Resolve qualifiedName first, then make single call to `memberBuilder.type(AstType.dynamicType(qualifiedName))`

Reply:
Suggest the following change to replace `if-else`.
```
String qualifiedTypeName = qualifiedNamesByLocalName.getOrDefault(typeName, typeName);
memberBuilder.type(AstType.dynamicType(qualifiedTypeName));
```